tests/test_lib_bfs.h

```diff
@@ -0,0 +1,49 @@
+/**
```

Comment:
should this file be under `tests/bfs`?

Reply:
If we would like to further have subdirectories for gtests I can move that under tests/bfs, right now since it is the only unit test we have I just kept it under tests.